src/writer.js

```diff
@@ -157,8 +156,8 @@ define([
     };
 
     var createBindList = function (form, xmlWriter) {
-        form.tree.walk(function (mug, nodeID, processChildren) {
-            if(mug && !mug.options.isControlOnly) {
+        form.dataTree().walk(function (mug, nodeID, processChildren) {
```

Comment:
The writer should probably construct the data tree once and pass it into `createDataBlock` as well as this function.